### PR TITLE
Toolbar stay hides on new Screen

### LINKED ISSUE #8052: Toolbar should stay hidden when the user hides it
Pressing <kbd>CTRL</kbd> + <kbd>T</kbd> hides the toolbar (when shown), but when navigating most screens it shows back and that makes the shortcut useless in this sense.

This may be also good for stable users who think the lazer ui has many elements on screen or just use shortcuts. Most overlays on the toolbar can be accessed using shortcuts already.

### PR DESCRIPTION
**Pull request by Volnetiks**
Closes #8052.

### CODE REVIEW

**Comment by mcendu** [comment]
Please add “closes #8052”

File: .idea/.idea.osu.Desktop/.idea/discord.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mcendu) [inline .idea/.idea.osu.Desktop/.idea/discord.xml]: Why add files?

Comment (Volnetiks) [reply .idea/.idea.osu.Desktop/.idea/discord.xml]: Uh, idk, I've never added it tbh lmao, I don't know where does it come from

Comment (Volnetiks) [reply .idea/.idea.osu.Desktop/.idea/discord.xml]: It comes from my discord RPC, should i remove the plugin?

Comment (peppy) [reply .idea/.idea.osu.Desktop/.idea/discord.xml]: you should not commit this. add it to your global .gitignore

Comment (Volnetiks) [reply .idea/.idea.osu.Desktop/.idea/discord.xml]: Okay, i will do this

File: osu.Game/Overlays/Toolbar/Toolbar.cs [hunk 2]
```diff
@@ -79,7 +79,7 @@ private void load(OsuGame osuGame, Bindable<RulesetInfo> parentRuleset)
                         //{
                         //    Icon = FontAwesome.Solid.search
                         //},
-                        userButton = new ToolbarUserButton(),
+                        userButton = new ToolbarUserButton(this),
```
Comment (bdach) [inline osu.Game/Overlays/Toolbar/Toolbar.cs]: What is the meaning of these changes? You can't be doing things like this in a commit with the message of `Updated .gitignore`.

File: .idea/.idea.osu.Desktop/.idea/discord.xml (hunk 1 above)
Comment (Volnetiks) [reply .idea/.idea.osu.Desktop/.idea/discord.xml]: Done, i've added it to the .gitignore, i will remove it from my fork

File: .gitignore [hunk 3]
```diff
@@ -334,3 +334,6 @@ inspectcode
 
 # BenchmarkDotNet
 /BenchmarkDotNet.Artifacts
+
+# Discord Plugin Integration
+.idea/.idea.osu.Desktop/.idea/discord.xml
```
Comment (bdach) [inline .gitignore]: I think what @peppy intended is to edit your *global `.gitignore` on your actual local machine*, not this repository's. Git can have per-machine ignores that span all repositories.

**Review by bdach — CHANGES_REQUESTED** [review]
Solution is incorrect, does not pass code style (as indicated by CI) and has no tests attached.

Also as a side note in general you do not want to be PRing from `master`. Make a branch instead.

File: osu.Game/Overlays/Toolbar/Toolbar.cs (hunk 2 above)
Comment (Volnetiks) [reply osu.Game/Overlays/Toolbar/Toolbar.cs]: Oh, i started working on an other issue, and i've tried something, but i added it in my commit gitignore, i will change it rn

File: .gitignore (hunk 3 above)
Comment (Volnetiks) [reply .gitignore]: Uh, i've never tried this, lemme check that

File: osu.Game/OsuGame.cs [hunk 4]
```diff
@@ -947,7 +947,7 @@ protected virtual void ScreenChanged(IScreen current, IScreen newScreen)
 
                 if (newOsuScreen.HideOverlaysOnEnter)
                     CloseAllOverlays();
-                else
+                else if(Toolbar.State.Value == Visibility.Visible)
```
[the remark this answers is not in the document]
Comment (Volnetiks) [reply osu.Game/OsuGame.cs]: Okay, i will try to find an other way for this issue

**Comment by peppy** [comment]
As the logic in this doesn't work and you have committed many unrelated changes, I'm going to close this. Please make tests covering changed logic if you attempt this again.

**Pull request closed without merging**